apps/aecore/src/aec_conductor.erl

```diff
@@ -576,26 +580,56 @@ erase_worker(Pid, #state{workers = Workers} = State) ->
 %%% Miner instances handling
 
 init_miner_instances(State) ->
-    MinerInstances = [{N, available} || N <- lists:seq(0, aec_pow_cuckoo:get_miner_instances() - 1)],
+    MinerConfigs   = aec_pow_cuckoo:get_miner_configs(),
+    MinerInstances = create_miner_instances(MinerConfigs),
     State#state{miner_instances = MinerInstances}.
 
+create_miner_instances(MinerConfigs) when is_list(MinerConfigs) ->
+    {MinerInstances, _} =
+        lists:foldl(
+          fun(MinerConfig, {Acc, Id}) ->
+                  {Instances, NextId} = create_miner_instances(MinerConfig, Id),
+                  {Instances ++ Acc, NextId}
+          end, {[], 0}, MinerConfigs),
+    MinerInstances.
+
+create_miner_instances(MinerConfig, FirstId) ->
+    case aec_pow_cuckoo:get_addressed_instances(MinerConfig) of
+        undefined ->
+            {[create_miner_instance(FirstId, undefined, MinerConfig)], FirstId + 1};
+        AddressedInstances when is_list(AddressedInstances) ->
+            lists:foldl(
+              fun(AddressedInstance, {Acc, Id}) ->
+                      {[create_miner_instance(Id, AddressedInstance, MinerConfig) | Acc], Id + 1}
+              end, {[], FirstId}, AddressedInstances)
+    end.
+
+create_miner_instance(Id, Instance, Config) ->
+    #miner_instance{id       = Id,
+                    nonce    = aec_pow:pick_nonce(),
```

Comment:
We are just trusting randomness for not rechecking nonces more than once, since we bind the nonce to the instance.

Wouldn't it make sense to keep the nonces in the candidate?

Reply:
I moved nonces to the miner as in my initial idea the whole miner config wasn't kept in the conductor. There was only some reference of the miner, and the `repeats` param of the particular miners was unknown (in the end this "miner references" idea didn't play well with parsing array of elements from yaml config..).

But now, since we have the whole miner config (together with `repeats` param) in the conductor, it is feasible, and I can change that 👍 There should not be much difference though, randomness should be good enough, given how ofter we should receive new candidates on top of key/micro blocks?!

Reply:
I agree that it probably wouldn't make a practical difference, but it would make sense to eliminate the possibility of getting the same nonce in different miners, since it is not a costly operation.

I could imagine a later situation where the randomness could be compromised since we are not explicitly seeding the random process, which in turn could lead to all miners working on the same nonce range. It seems safe now, though.